### PR TITLE
Updated workflow documentation

### PR DESCRIPTION
**Pull request by HasiniSama**

### CODE REVIEW

File: docs/resource-propagation.md [hunk 1]
```diff
@@ -1,13 +1,49 @@
 # Resource Propagation across Multiple Environments
 
-This section describes how the IAM-CTL tool can be used to continously propagate resources across multiple environments.
+This repository provides instructions on how to continuously propagate resources across multiple environments using the IAM-CTL tool.
```
Comment (divyaamunugama) [inline docs/resource-propagation.md]:
```suggestion
This repository provides instructions on continuously propagating resources across multiple environments using the IAM-CTL tool.
```

File: docs/resource-propagation.md [hunk 2]
```diff
@@ -1,13 +1,49 @@
...
-2. Create a separate config folder for each environment with the relevant server details and other configurations.
-3. Export resources from the lower environment to a local directory, which would be the common resource configuration directory for all environments.
-4. If there are environment specific variables, add keyword placeholders to the exported files, and add the relevant keyword mapping to the tool configs.
-5. Use the import command to deploy the resources from the local directory to higher environments.
-6. When new changes are added to the lower environment, export the resources again to update the local resource configurations and import them back to higher environments.
+## Recommended Workflow
 
+To manage resources when multiple environments exist, follow this recommended workflow:
+
+1. **Set Up WSO2 Identity Server Instances**: Have separate WSO2 Identity Server instances for each environment, hosted somewhere accessible by the public.
```
Comment (divyaamunugama) [inline docs/resource-propagation.md]:
```suggestion
1. **Set up WSO2 Identity Server instances**: Have separate WSO2 Identity Server instances for each environment, hosted somewhere accessible by the public.
```

File: docs/resource-propagation.md [hunk 3]
```diff
@@ -1,13 +1,49 @@
...
+
+2. **Create Management Application**: Create an application with the Management Application enabled in the target Identity Server. Update the OAuth inbound authentication configuration with a dummy callback URL. Make note of the Client ID and Client Secret for later use. *(Automation will be implemented in the future)*
```
Comment (divyaamunugama) [inline docs/resource-propagation.md]:
```suggestion
2. **Create a management application**: Create an application with the Management Application capabilities enabled in the target Identity Server. Update the OAuth inbound authentication configuration with a dummy callback URL. Make note of the Client ID and Client Secret for later use. *(Automation will be implemented in the future)*
```

File: docs/resource-propagation.md [hunk 4]
```diff
@@ -1,13 +1,49 @@
...
+
+3. **Git Repository Setup**:
```
Comment (divyaamunugama) [inline docs/resource-propagation.md]:
```suggestion
3. **Git repository setup**:
```

File: docs/resource-propagation.md [hunk 5]
```diff
@@ -1,13 +1,49 @@
...
+
+   - Create a Git repository with branches dedicated to each environment. Use the environment name (same as the branch name) as an environment variable with the key `ENV` for each branch. This is where the resource config files will reside for that environment.
+
+   - **Manual Approach**:
```
Comment (divyaamunugama) [inline docs/resource-propagation.md]:
```suggestion
   - **Manual approach**:
```

File: docs/resource-propagation.md [hunk 6]
```diff
@@ -1,13 +1,49 @@
...
+     - Clone the repository to your local machine.
+     - Download the latest release of the IAM-CTL tool from [GitHub Releases](https://github.com/wso2-extensions/identity-tools-cli/releases) and unzip it.
+     - Run the following command, replacing `<path to the cloned repo folder>` with the appropriate path: 
+       ```
+       iamctl setupCLI -d "<path to the cloned repo folder>"
+       ```
+     - Create subfolders for each environment in the `config` folder and configure each `serverConfig.json` file with the relevant server details and other configurations.
+     - *Note: Variables of `serverConfig.json` can also be added as environment values. Refer to the [documentation](<LINK>) for more information on loading server configurations from environment variables.*
```
Comment (divyaamunugama) [inline docs/resource-propagation.md]: Add the link here

File: docs/resource-propagation.md [hunk 7]
```diff
@@ -1,13 +1,49 @@
...
+
+   - **Workflow Automation**:
```
Comment (divyaamunugama) [inline docs/resource-propagation.md]:
```suggestion
   - **Workflow automation**:
```

File: docs/resource-propagation.md [hunk 8]
```diff
@@ -1,13 +1,49 @@
...
+     - Set up the environment variable. Refer to the [documentation](<LINK>) for more information on loading server configurations from environment variables.
```
Comment (divyaamunugama) [inline docs/resource-propagation.md]: Add link

File: docs/resource-propagation.md [hunk 9]
```diff
@@ -1,13 +1,49 @@
...
+     - Add a GitHub Action with the custom action `@action/setup` (replace with the correct action reference).
+     - Trigger the action.
+     - An automatically generated `config` folder will be added to the repository.
+
+4. **Export Resources**:
```
Comment (divyaamunugama) [inline docs/resource-propagation.md]:
```suggestion
4. **Export resources**:
```

File: docs/resource-propagation.md [hunk 10]
```diff
@@ -1,13 +1,49 @@
...
+   - Export resources from the lower environment to the lower environment branch using another GitHub Action with the custom action `@action/export`.
```
Comment (divyaamunugama) [inline docs/resource-propagation.md]:
```suggestion
   - Export resources from the lower environment to the lower environment branch using another GitHub action with the custom action `@action/export`.
```

File: docs/resource-propagation.md [hunk 11]
```diff
@@ -1,13 +1,49 @@
...
+   - A pull request will be created with the changes. Review and verify the correctness of the propagation, and then merge the pull request.
+
+5. **Environment-specific Variables**:
```
Comment (divyaamunugama) [inline docs/resource-propagation.md]:
```suggestion
5. **Environment-specific variables**:
```

File: docs/resource-propagation.md [hunk 12]
```diff
@@ -1,13 +1,49 @@
...
+   - If there are environment-specific variables, add keyword placeholders to the exported files and add the relevant keyword mapping to the tool configurations.
+
+6. **Merge to Higher Environment Branch**:
```
Comment (divyaamunugama) [inline docs/resource-propagation.md]:
```suggestion
6. **Merge to higher environment branch**:
```

File: docs/resource-propagation.md [hunk 13]
```diff
@@ -1,13 +1,49 @@
...
+   - Merge the lower environment branch to the higher environment branch. *(Automation can be implemented to automatically merge the pull request from the lower environment to the higher environment branch.)*
+
+7. **Import Resources**:
```
Comment (divyaamunugama) [inline docs/resource-propagation.md]:
```suggestion
7. **Import resources**:
```

File: docs/resource-propagation.md [hunk 14]
```diff
@@ -1,13 +1,49 @@
...
+   - Use the import command to deploy the resources from the repository to the higher environment using another GitHub Action with the custom action `@action/import`.
+   - *(Automation can also be implemented by triggering the Git action automatically upon merging to the branch.)*
+
+8. **Update and Import New Changes**:
```
Comment (divyaamunugama) [inline docs/resource-propagation.md]:
```suggestion
8. **Update and import new changes**:
```

File: docs/resource-propagation.md [hunk 15]
```diff
@@ -1,13 +1,49 @@
...
+
+   - **Manual approach**:
+     - Clone the repository to your local machine.
+     - Download the latest release of the IAM-CTL tool from [GitHub Releases](https://github.com/wso2-extensions/identity-tools-cli/releases) and unzip it.
+     - Run the following command, replacing `<path to the cloned repo folder>` with the appropriate path: 
+       ```
+       iamctl setupCLI -d "<path to the cloned repo folder>"
+       ```
+     - Create subfolders for each environment in the `config` folder and configure each `serverConfig.json` file with the relevant server details and other configurations.
+     - *Note: Variables of `serverConfig.json` can also be added as environment values. Refer to the [documentation](<LINK>) for more information on loading server configurations from environment variables.*
```
Comment (ImalshaG) [inline docs/resource-propagation.md]:
```suggestion
     - *Note: Variables of `serverConfig.json` can also be added as environment values. Refer to the [documentation](https://github.com/wso2-extensions/identity-tools-cli/blob/master/docs/cli-mode.md#load-server-configurations-from-environment-variables) for more information on loading server configurations from environment variables.*
```

File: docs/resource-propagation.md [hunk 16]
```diff
@@ -1,13 +1,49 @@
...
+
+   - **Workflow automation**:
+     - Set up the environment variable. Refer to the [documentation](<LINK>) for more information on loading server configurations from environment variables.
```
Comment (ImalshaG) [inline docs/resource-propagation.md]:
```suggestion
     - Set up the environment variable. Refer to the [documentation](https://github.com/wso2-extensions/identity-tools-cli/blob/master/docs/cli-mode.md#load-server-configurations-from-environment-variables) for more information on loading server configurations from environment variables.
```

**Review by divyaamunugama — APPROVED** [review]

**Review by ImalshaG — APPROVED** [review]

**Pull request merged**